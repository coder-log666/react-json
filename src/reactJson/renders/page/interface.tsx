import { SchemaClassName, SchemaCollection } from "../../interface/schema";
import { BaseSchema, RendererProps } from "../interface";

export interface PageSchema extends BaseSchema {
    /**
     * 指定为 page 渲染器。
     */
    type: "page";
  
    /**
     * 页面标题
     */
    title?: string;
  
    /**
     * 页面副标题
     */
    subTitle?: string;

    /**
     * 内容区域
     */
    body?: SchemaCollection;
  
    /**
     * 内容区 css 类名
     */
    bodyClassName?: SchemaClassName;
  
    /**
     * 配置 header 容器 className
     */
    headerClassName?: SchemaClassName;
}

  
export interface PageProps
  extends RendererProps,
    Omit<PageSchema, "type" | "className"> {
  data: any;
//   store: IServiceStore;
  location?: Location;
}