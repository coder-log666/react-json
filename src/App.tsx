import React, {useContext} from "react";

import RenderContext from './context'

export default function App() {
    const jsonRender: any = useContext(RenderContext);
    let a = jsonRender({
        type: 'page',
        className: 'jsonPage',
        title: 'page标题',
        subTitle: 'page副标题',
        bodyClassName: 'page-body-className',
        headerClassName: 'page-header-className',
        body: {
            type: 'page'
        }
    })
    console.log(11111)
    return a
}