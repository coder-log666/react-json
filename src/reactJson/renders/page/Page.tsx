import { Renderer } from "../../core/factory";
import React from "react";
import { PageProps } from "./interface";

@Renderer({
    test: /(?:^|\/)page$/,
    name: "page"
  })
  export class PageRenderer extends React.Component<PageProps> {
    render(): React.ReactNode {
      const {className, title, subTitle, headerClassName, body, bodyClassName} = this.props
        return (<div className={"page" + ` ${className}`}>
            <div className={"page-header" + ` ${headerClassName}`}>
              <div className="title">
                { title }
              </div>
              <div className="subtitle">
                { subTitle }
              </div>
            </div>
            <div className={"page-body" + ` ${bodyClassName}`}>
              {'page-body'}
            </div>
        </div>)
    }
  }