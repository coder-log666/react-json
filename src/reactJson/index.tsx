import {
    render,
    Renderer,
  } from './core/factory';
 

  import './renders/page/Page';

  export {
    render,
    Renderer
  };