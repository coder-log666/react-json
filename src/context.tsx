import React from 'react'
import {render} from './reactJson/index'

const RenderContext = React.createContext({render: render});

export default RenderContext