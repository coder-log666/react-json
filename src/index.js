import React from "react";
import ReactDOM from 'react-dom/client';
import App from './App'
import {render} from './reactJson/index'

const root = ReactDOM.createRoot(document.getElementById('root'));
import RenderContext from './context'

root.render(
  <React.StrictMode>
    <RenderContext.Provider value={render}>
        <App />
    </RenderContext.Provider>
  </React.StrictMode>
);


