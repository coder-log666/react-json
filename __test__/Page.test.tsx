import React from "react";
import * as renderer from 'react-test-renderer';
import {render as amisRender} from '../src/reactJson';
import {PageRenderer} from '../src/reactJson/renders/page/Page';

test('Renderer:Page', () => {
  const component = renderer.create(
    amisRender({
      type: 'page'
    })
  );
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});