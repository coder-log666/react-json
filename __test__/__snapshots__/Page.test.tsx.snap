// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderer:Page 1`] = `
<div>
  <div>
    page
  </div>
</div>
`;
